sdk/attestation/Azure.Security.Attestation/src/AttestationToken.cs

```diff
@@ -645,7 +649,11 @@ private static string GenerateSecuredJsonWebToken(BinaryData body, AttestationTo
             AsymmetricAlgorithm signer;
             if (signingKey.Certificate.HasPrivateKey)
             {
+#if NET6_0_OR_GREATER
+                signer = (AsymmetricAlgorithm)signingKey.Certificate.GetRSAPrivateKey() ?? signingKey.Certificate.GetDSAPrivateKey();
+#else
                 signer = signingKey.Certificate.PrivateKey;
```

Comment:
See: [SYSLIB0028: X509Certificate2.PrivateKey is obsolete](https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0028)